src/webui/api/serialize/serialize_torrent.cpp

```diff
@@ -166,6 +166,7 @@ QVariantMap serialize(const BitTorrent::Torrent &torrent)
         {KEY_TORRENT_COMMENT, torrent.comment()},
         {KEY_TORRENT_PRIVATE, (torrent.hasMetadata() ? torrent.isPrivate() : QVariant())},
         {KEY_TORRENT_TOTAL_SIZE, torrent.totalSize()},
-        {KEY_TORRENT_HAS_METADATA, torrent.hasMetadata()}
+        {KEY_TORRENT_HAS_METADATA, torrent.hasMetadata()},
+        {KEY_TORRENT_PERCENT_SELECTED, torrent.hasMetadata() ? ((static_cast<qreal>(torrent.wantedSize()) * 100) / torrent.totalSize()) : -1},
```

Comment:
I strongly disapprove of adding such fields, the value of which is trivially calculated from the values of existing primary fields.